dune/gdt/operators/operator.hh

```diff
@@ -0,0 +1,341 @@
+               << "} local {element|intersection} operators to jacobian_op ..." << std::endl;
+    const auto source_function = make_discrete_function(source_space_, source_vector);
+    const auto append_local_operator_to_jacobian = [&](const auto& list_of_data) {
+      for (const auto& data : list_of_data) {
+        const auto local_operator = data.first->with_source(source_function);
+        const auto& filter = *data.second;
+        jacobian_op.append(*local_operator, source_vector, parameter, filter);
+      }
+    };
+    append_local_operator_to_jacobian(element_data_);
+    append_local_operator_to_jacobian(intersection_data_);
+  } // ... jacobian(...)
+
+  /// \}
+  /// \name These methods allow to append local operators
+  ///\{
+
+  ThisType& operator+=(const LocalElementOperatorType& local_operator)
+  {
+    LOG_(info) << "+=(local_element_operator=" << &local_operator << ")" << std::endl;
+    this->extend_parameter_type(local_operator.parameter_type());
+    linear_ = linear_ && local_operator.linear();
+    element_data_.emplace_back(local_operator.copy(), new ApplyOnAllElements());
+    return *this;
+  }
+
+  ThisType&
+  operator+=(const std::pair<const LocalElementOperatorType&, const ElementFilterType&>& local_operator__filter)
+  {
+    const auto& local_operator = local_operator__filter.first;
+    const auto& filter = local_operator__filter.second;
+    LOG_(info) << "+=(local_element_operator=" << &local_operator << ", element_filter=" << &filter << ")" << std::endl;
+    this->extend_parameter_type(local_operator.parameter_type());
+    linear_ = linear_ && local_operator.linear();
+    element_data_.emplace_back(local_operator.copy(), filter.copy());
+    return *this;
+  } // ... operator+=(...)
+
+  ThisType& operator+=(const LocalIntersectionOperatorType& local_operator)
+  {
+    LOG_(info) << "+=(local_intersection_operator=" << &local_operator << ")" << std::endl;
+    this->extend_parameter_type(local_operator.parameter_type());
+    linear_ = linear_ && local_operator.linear();
+    intersection_data_.emplace_back(local_operator.copy(), new ApplyOnAllIntersections());
+    return *this;
+  }
+
+  ThisType& operator+=(
+      const std::pair<const LocalIntersectionOperatorType&, const IntersectionFilterType&>& local_operator__filter)
+  {
+    const auto& local_operator = local_operator__filter.first;
+    const auto& filter = local_operator__filter.second;
+    LOG_(info) << "+=(local_intersection_operator=" << &local_operator << ", intersection_filter=" << &filter << ")"
+               << std::endl;
+    this->extend_parameter_type(local_operator.parameter_type());
+    linear_ = linear_ && local_operator.linear();
+    intersection_data_.emplace_back(local_operator.copy(), filter.copy());
+    return *this;
+  } // ... operator+=(...)
+
+  /// \}
+
+  const std::list<std::pair<std::unique_ptr<LocalElementOperatorType>, std::unique_ptr<ElementFilterType>>>&
```

Comment:
auto return?

Reply:
Do we have a view on this (when to use auto)? The old 'auto is easier' vs. 'I can see what the type is' argument. Currently, I use `auto` when it *gets too complicated*, so this could apply here. Do we want to use it more often in general?

Reply:
Maybe use something like "more than one nesting level --> use auto" as a rule of thumb? If one can just use a type alias I still think it unnecessary/silly do use auto, yeah.